### ISSUE TITLE
strftime precision against combinated formats (ex. %F, %r, ...)

### ISSUE THREAD

**Issue by dmikurube** [jruby/jruby#6169]
`Time.strftime` has some combinated formats, such as `%F` equal to `%Y-%m-%d`, and `%r` equal to `%I:%M:%S %p`.

Users can apply options even for those combinated formats, for example `%20F`. But the behavior is different between CRuby and JRuby.

**Environment Information**

* Confirmed with JRuby 9.1.15.0 and 9.2.10.0.
* Mac OS X 10.13. (The OS does not sound related, though.)

**Expected Behavior**

This is CRuby's behavior. (2.4, 2.5, 2.6, 2.7)

```
irb(main):001:0> require 'time'
=> true
irb(main):002:0> Time.now.strftime("%20F")
=> "          2020-04-06"
```

**Actual Behavior**

JRuby is working like :
```
irb(main):001:0> require 'time'
=> true
irb(main):002:0> Time.now.strftime("%20F")
=> "00000000000000002020-04-06"
```

**Comment by headius** [comment]
Ok this is broken because we look at the first element to print after the precision, and it's a numeric year, so we assume it should be numeric padding. I'm looking into the right way to indicate that this pattern should remain text padded with spaces.

**Comment by headius** [comment]
I've pushed a kinda ugly fix in #6228 but I feel like perhaps the strftime parser should be handling this.
